client/app/scripts/charts/nodes-chart-elements.js

```diff
@@ -15,18 +14,12 @@ class NodesChartElements extends React.Component {
           selectedScale={props.selectedScale}
           isAnimated={props.isAnimated} />
         <NodesChartNodes
-          layoutNodes={props.completeNodes}
+          layoutNodes={props.layoutNodes}
           selectedScale={props.selectedScale}
           isAnimated={props.isAnimated} />
       </g>
     );
   }
 }
 
-function mapStateToProps(state, props) {
-  return {
-    completeNodes: completeNodesSelector(state, props)
-  };
-}
-
-export default connect(mapStateToProps)(NodesChartElements);
+export default connect()(NodesChartElements);
```

Comment:
Nice! I did the same with all the other components as well, made them `PureComponent` instead of using empty Redux `connect()`.